### PR TITLE
Add more debug, and timeout on DHCP rather than silently waiting forever

### PR DESCRIPTION
**Pull request by nmatthews-witekio**
Split DHCP timeout into sections and add a delay.

The debug is helpful to track progress -- having a timeout is also useful -- rather than waiting indefinitely without any output (if there's a problem).

Not sure whether splitting the timeout into sections is necessary -- but it should give other threads a chance to update themselves and maybe IP related values, too.

### CODE REVIEW

**Comment by nbeck-witekio** [comment]
This error checking is an improvement. We should free the DHCP resource too when failures occur and when it is finished with using nx_dhcp_delete()

File: samples/ck-rx65n/basic-sample/src/main.c [hunk 1]
```diff
@@ -343,34 +343,60 @@ ULONG   gateway_address = 0;
...
+    ULONG   dns_server_address[3]; // IPv6 needs more than one ULONG
+    UINT    dns_server_address_size = sizeof(dns_server_address);
 
     /* Create a DNS instance for the Client.  Note this function will create
        the DNS Client packet pool for creating DNS message packets intended
        for querying its DNS server. */
     status = nx_dns_create(&dns_0, &ip_0, (UCHAR *)"DNS Client");
     if (status)
     {
+        printf("nx_dns_create failed: status = %d\r\n", status);
```
Comment (nik-markovic) [inline samples/ck-rx65n/basic-sample/src/main.c]: *status* is unsigned, it should be %u, or even better "0x%x" to be consistent with other status checks.

Needs to be updated other printfs.

I'm curuous though how did this even compile without warnings maybe make sure that we aren't ignoring compiler warnings in flags?

File: samples/ck-rx65n/basic-sample/src/main.c [hunk 2]
```diff
@@ -343,34 +343,60 @@ ULONG   gateway_address = 0;
...
+        printf("retry nx_ip_status_check %d\r\n", i);
+        status = nx_ip_status_check(&ip_0, NX_IP_ADDRESS_RESOLVED, &actual_status, 1000);
+    }
+    if(status != NX_SUCCESS)
+    {
+        printf("nx_ip_status_check failed: %d\r\n", status);
+        return;
+    }
+    printf("nx_ip_status_check ok\r\n");
+    printf("...DHCP Finished\r\n");
```
Comment (nik-markovic) [inline samples/ck-rx65n/basic-sample/src/main.c:396]: Let's please not spam the logs. Ether one or the other message.

File: samples/ck-rx65n/basic-sample/src/main.c [hunk 3]
```diff
@@ -343,34 +343,60 @@ ULONG   gateway_address = 0;
...
+    int status;
 
     printf("DHCP In Progress...\r\n");
 
     /* Create the DHCP instance.  */
-    nx_dhcp_create(&dhcp_0, &ip_0, "DHCP Client");
+    status = nx_dhcp_create(&dhcp_0, &ip_0, "DHCP Client");
+    if(status != NX_SUCCESS) {
+        printf("nx_dhcp_create failed: %d\r\n", status);
+        return;
```
Comment (nik-markovic) [inline samples/ck-rx65n/basic-sample/src/main.c]: I'm not 100% sure about this, but I am concerned about returning from a failure. If there's a problem, nothing will work anyways. It is safrer to stay lock the thread in this function. We can do tx_thread_sleep() in a forever loop. The sleep should also allow printfs to flush.

File: samples/ck-rx65n/basic-sample/src/main.c [hunk 4]
```diff
@@ -343,34 +343,60 @@ ULONG   gateway_address = 0;
...
+    }
+    printf("nx_dhcp_create ok\r\n");
```
Comment (nik-markovic) [inline samples/ck-rx65n/basic-sample/src/main.c:367]: We should not be spamming the log and prininging all these status messages unless we are tracing. Please remove successful messages from the logs. You can leave important ones in.

It should be safe to assume that the outcome is success (print the last status message if needed) unlesss a stage failed, which will print the message in the log.

**Review by nik-markovic — CHANGES_REQUESTED** [review]

**Comment by nmatthews-witekio** [comment]
Included and rebased in https://github.com/avnet-iotconnect/iotc-azurertos-sdk/pull/52

**Pull request closed without merging**